sandbox/extract-imagery-metadata_emp.R

```diff
@@ -0,0 +1,591 @@
+# this script again, and when it extracts the metadata for each EXIF dataset, your additional
+# metadata should be included.
+
+
+
+
+
+#### Image-level metadata CSV file (one file per dataset) ####
+
+# Output file: imagery-metadata-dev/extracted-metadata/image-level-metadata/image-metadata_<dataset_id>.csv
+
+exif_file = exif_files[1]
+exif = prep_exif(exif_file)
+
+#### First need to make sure each exif file has the right columns for the following functions. If column(s) are missing, need to add columns of blanks or NAs ####
+
+#### dataset ID ####
+
+# BEGIN FUNCTION CODE
+
+dataset_id = exif$dataset_id
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_dataset_id = function (exif) {
+
+  dataset_id = exif$dataset_id
+
+  return(dataset_id)
+
+}
+
+# test function on exif data
+
+dataset_id = extract_dataset_id (exif)
+
+dataset_id
+
+#### datatime_local (Format: YYYYMMDD HHMMSS (local time zone, 24 hr)) ####
+
+# BEGIN FUNCTION CODE
+
+datatime_local = (exif$DateTimeOriginal)
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_datatime_local = function (exif) {
+
+  datatime_local = (exif$DateTimeOriginal)
+
+  return(datatime_local)
+
+}
+
+# test function on exif data
+
+datatime_local = extract_datatime_local(exif)
+
+datatime_local
+
+#### lat (Format: dd.dddddddd (EPSG:4326)) ####
+
+# BEGIN FUNCTION CODE
+
+exif_coordinates <- data.frame(exif$X, sf::st_coordinates(exif[,1], st_coordinates(exif[,2])))
+
+lat = exif_coordinates$Y
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_lat = function (exif) {
+
+  exif_coordinates <- data.frame(exif$X, sf::st_coordinates(exif[,1], st_coordinates(exif[,2])))
+
+  lat = exif_coordinates$Y
+
+  return(lat)
+}
+
+# test function on exif data
+
+lat = extract_lat(exif)
+lat
+
+#### lon (Format: dd.dddddddd (EPSG:4326)) ####
+
+# BEGIN FUNCTION CODE
+
+exif_coordinates <- data.frame(exif$X, sf::st_coordinates(exif[,1], st_coordinates(exif[,2])))
+
+lon = exif_coordinates$X
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_lon = function (exif) {
+
+  exif_coordinates <- data.frame(exif$X, sf::st_coordinates(exif[,1], st_coordinates(exif[,2])))
+
+  lon = exif_coordinates$X
+
+  return(lon)
+}
+
+# test function on exif data
+
+lon = extract_lon(exif)
+lon
+
+#### rtk_fix (Format: True/False. Use EXIF RTKFlat) ####
+
+# goal: return TRUE if there is a RtkFlag value of 50, and FALSE otherwise (i.e. other RtkFlag value, or none). IMPORTANT NOTE!!!!!! THIS FUNCTION ASSUMES DJI DRONES WERE USED. OTHER DRONE MAKES WILL REQUIRE AN UPDATED FUNCTION.
+
+# BEGIN FUNCTION CODE
+
+rtk_fix = ifelse (exif$RtkFlag == 50, "TRUE", ifelse (exif$RtkFlag == 0, "FALSE", "Other"))
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_rtk_fix = function(exif) {
+
+  exif["RtkFlag"[!("RtkFlag" %in% colnames(exif))]] = NA
+
+  rtk_fix = ifelse (exif$RtkFlag == 50, "TRUE", ifelse (exif$RtkFlag == 0, "FALSE", "Other"))
```

Comment:
All good points. I originally put in "other" because there are a bunch DJI values that can be returned in this field. Here are some of those: 
0 - Failed to position 
16 - Single point positioning (meter-level accuracy) 
32~49 - Floating point solution positioning (decimeter-level to meter-level accuracy) 
50 - Fixed solution positioning (centimeter-level accuracy)

So of course 50 is true and 0 (or a missing column) is false, but what about the meter to decimeter level accuracy? Happy to add in other values from this list that return true or false, let me know!

Reply:
50 (fixed) is the only one that is true RTK-level precision; I think we are best off assuming all the rest are of equivalent "non-RTK presicion". So let's go wtih 50 being TRUE and anything else (including missing) being FALSE -- until we learn of a drone that uses a different code for "fixed".